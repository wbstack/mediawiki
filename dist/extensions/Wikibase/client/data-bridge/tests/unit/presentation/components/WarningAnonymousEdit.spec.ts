import MessageKeys from '@/definitions/MessageKeys';
import EventEmittingButton from '@/presentation/components/EventEmittingButton.vue';
import WarningAnonymousEdit from '@/presentation/components/WarningAnonymousEdit.vue';
import { shallowMount, config } from '@vue/test-utils';

config.renderStubDefaultSlot = true;

describe( 'WarningAnonymousEdit', () => {
	it( 'matches the snapshot', () => {
		const $messages = {
			KEYS: MessageKeys,
			get: jest.fn().mockReturnValue( 'Some <abbr>HTML</abbr>.' ),
			getText: jest.fn().mockReturnValue( "Some 'text" ),
		};
		const wrapper = shallowMount( WarningAnonymousEdit, {
			propsData: {
				loginUrl: 'https://data-bridge.test/Login',
			},
			global: {
				mocks: { $messages },
			},
		} );

		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'maps proceed button click to proceed event', async () => {
		const wrapper = shallowMount( WarningAnonymousEdit, {
			propsData: {
				loginUrl: 'https://data-bridge.test/Login',
			},
		} );

		wrapper.findComponent( EventEmittingButton ).vm.$emit( 'click' );
		expect( wrapper.emitted( 'proceed' ) ).toHaveLength( 1 );
	} );
} );
