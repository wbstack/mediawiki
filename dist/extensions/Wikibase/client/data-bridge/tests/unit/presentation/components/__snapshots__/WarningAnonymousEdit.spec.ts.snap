// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningAnonymousEdit matches the snapshot 1`] = `
<div
  class="wb-db-warning-anonymous-edit"
>
  <h2
    class="wb-db-warning-anonymous-edit__heading"
  >
    Some 'text
  </h2>
  <icon-message-box-stub
    class="wb-db-warning-anonymous-edit__message"
    inline="false"
    type="warning"
  >
    <p
      class="wb-db-warning-anonymous-edit__message-text"
    >
      Some 
      <abbr>
        HTML
      </abbr>
      .
    </p>
  </icon-message-box-stub>
  <div
    class="wb-db-warning-anonymous-edit__buttons"
  >
    <event-emitting-button-stub
      class="wb-db-warning-anonymous-edit__proceed"
      disabled="false"
      message="Some 'text"
      newtab="false"
      preventdefault="true"
      size="M"
      squary="false"
      type="primaryProgressive"
    />
    <event-emitting-button-stub
      class="wb-db-warning-anonymous-edit__login"
      disabled="false"
      href="https://data-bridge.test/Login"
      message="Some 'text"
      newtab="false"
      preventdefault="false"
      size="M"
      squary="false"
      type="neutral"
    />
  </div>
</div>
`;
